components/ruby/lib/chef-licensing/license_key_fetcher.rb

```diff
@@ -183,6 +183,19 @@ def fetch
       (fetch_license_key_from_arg << fetch_license_key_from_env << @license_keys).flatten.uniq
     end
 
+    def warn_if_non_commercial_license
+      logger.debug "License Key fetcher - checking if licenses are commercial"
+      self.license = ChefLicensing.client(license_keys: license_keys)
```

Comment:
We can memoize the license variable here. Or have a check.

Reply:
Also can use https://github.com/chef/chef-licensing/pull/171 for api call improvement.

Reply:
Removing the additional method and hence the additional API call.